exp/api/v1alpha4/azuremanagedmachinepool_webhook.go

```diff
@@ -51,10 +52,33 @@ func (r *AzureManagedMachinePool) Default(client client.Client) {
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (r *AzureManagedMachinePool) ValidateCreate(client client.Client) error {
 	azuremanagedmachinepoollog.Info("validate create", "name", r.Name)
+	var allErrs field.ErrorList
+	// If AutoScaling is disabled, both MinCount and MaxCount should not be set
+	if (r.Spec.MaxCount != nil || r.Spec.MinCount != nil) && (r.Spec.EnableAutoScaling == nil || !*r.Spec.EnableAutoScaling) {
+		allErrs = append(allErrs,
+			field.Invalid(
+				field.NewPath("Spec", "EnableAutoScaling"),
+				r.Spec.EnableAutoScaling,
+				"MaxCount and MinCount cannot be set when AutoScaling is disabled"))
+	}
+
+	// If AutoScaling is enabled, both MinCount and MaxCount should be set
+	if r.Spec.EnableAutoScaling != nil && (r.Spec.MaxCount == nil || r.Spec.MinCount == nil) && *r.Spec.EnableAutoScaling {
+		allErrs = append(allErrs,
+			field.Invalid(
+				field.NewPath("Spec", "EnableAutoScaling"),
+				r.Spec.EnableAutoScaling,
+				"MaxCount and MinCount must be set when AutoScaling is enabled"))
+	}
+
+	if len(allErrs) != 0 {
+		return apierrors.NewInvalid(GroupVersion.WithKind("AzureManagedMachinePool").GroupKind(), r.Name, allErrs)
+	}
 	return nil
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
+//gocyclo:ignore
```

Comment:
```suggestion
```

Reply:
Please think of a more elegant way to do this :) 

I think cyclomatic complexity is configured to : 

- reduce the complexity of the code
- avoid cognitive burden
- reduce the bugginess in code

If it is breaking it means we should try and improve code.

Reply:
+1 

also pasting my slack comment for reference: https://kubernetes.slack.com/archives/CEX9HENG7/p1631747376132600?thread_ts=1631743825.132300&cid=CEX9HENG7